### ISSUE TITLE
Trojan +WS Fallback returns empty pages with `200` status code (instead of returning the real webpage)

### ISSUE THREAD

**Issue by CXwudi** [SagerNet/sing-box#878]
### Operating system

Linux

### System version

Whatever Linux used by the Docker Image [`gzxhwq/sing-box:dev`](https://hub.docker.com/r/gzxhwq/sing-box)

### Installation type

Original sing-box Command Line

### If you are using a graphical client, please provide the version of the client.

Client is not involved in this issue

### Version

<details>

``` console

🚀 docker compose run sing-box-server version
sing-box version 1.5.0-beta.7

Environment: go1.21.1 linux/amd64
Tags: with_gvisor,with_quic,with_wireguard,with_utls,with_reality_server,with_clash_api,with_acme,with_dhcp,with_ech     
Revision: 90957564eb0522fb785eb2955c7cea3716e68a45
CGO: disabled
```

</details>

### Description

After setting up a Trojan+WS server, the fallback is not returning the webpage, but an empty page with `200` status code.
![photo_5177095858989149320_y](https://github.com/SagerNet/sing-box/assets/17377423/f585a692-84e8-497c-b832-7079aaea0ffd)
![photo_5177095858989149321_y](https://github.com/SagerNet/sing-box/assets/17377423/0034357c-bc43-4e3c-8a74-2c5e8fb4f043)

I also noticed that by removing WS transport, the issue disappeared, and the fallback page can be displayed correctly.

I also noticed you said that [Transport is not supported by Trojan's Fallback](https://github.com/SagerNet/sing-box/issues/269#issuecomment-1336067296). If that so, you can simply close this issue but I hope that the documentation can be clarified.

### Reproduction

To help reproduce the issue, I made [a sample docker compose zip file](https://github.com/SagerNet/sing-box/files/12583966/sing-box-issue-reproducible.zip).

0. Install Docker and Docker Compose
1. Download the attachment and unzip the file
2. Open a terminal in the directory where the `compose.yml` file located
3. Run `docker compose up -d` to bring up the sing-box server with the fallback site
4. Open a browser and open http://localhost:8081 to check out the fallback site directly (which is just the default nginx page)
5. Open http://localhost:8080 to check out the fallback site through sing-box's trojan+WS, observer the issue appears

Optional step:

7. run `docker compose down` to stop the sing-box container, remove the transport section in the `conf/common_inbounds.json` file, run `docker compose up -d` again, see that http://localhost:8080 returns the fallback site normally

### Logs

<details>
Logs of Trojan+WS:

``` console

+0000 2023-09-12 08:02:23 INFO router: updated default interface eth0, index 18
+0000 2023-09-12 08:02:23 INFO inbound/mixed[mixed-in]: tcp server started at [::]:1080
+0000 2023-09-12 08:02:23 INFO inbound/trojan[trojan-in]: tcp server started at [::]:80
+0000 2023-09-12 08:02:23 INFO sing-box started (0.22s)
+0000 2023-09-12 08:02:33 INFO inbound/trojan[trojan-in]: fallback connection to fake-site-entry:80
+0000 2023-09-12 08:02:33 INFO outbound/direct[direct]: outbound connection to fake-site-entry:80
+0000 2023-09-12 08:02:34 INFO inbound/trojan[trojan-in]: fallback connection to fake-site-entry:80
+0000 2023-09-12 08:02:34 INFO outbound/direct[direct]: outbound connection to fake-site-entry:80

```

Logs of Trojan (without WS):

``` console

+0000 2023-09-12 08:15:07 INFO router: updated default interface eth0, index 23
+0000 2023-09-12 08:15:07 INFO inbound/mixed[mixed-in]: tcp server started at [::]:1080
+0000 2023-09-12 08:15:07 INFO inbound/trojan[trojan-in]: tcp server started at [::]:80
+0000 2023-09-12 08:15:07 INFO sing-box started (0.00s)
+0000 2023-09-12 08:15:11 INFO [2405718809 0ms] inbound/trojan[trojan-in]: inbound connection from 172.21.0.1:54644
+0000 2023-09-12 08:15:11 INFO [2405718809 3ms] inbound/trojan[trojan-in]: fallback connection to fake-site-entry:80
+0000 2023-09-12 08:15:11 INFO [2405718809 4ms] outbound/direct[direct]: outbound connection to fake-site-entry:80

```

</details>

**Comment by nekohasekai** [comment]
Yes, there is no such support. There is no evidence that GFW proactively initiates HTTPS requests and blocks based on responses, and such functionality serves no purpose other than low-performance traffic multiplexing.

**Comment by CXwudi** [comment]
I see

It would be better if you could clarify it in the documentation. Say something like Trojan fallback is not supported when using any Transport. Rather than just saying no evidence of GFW initiating HTTPS requests
